src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -100,6 +100,8 @@ def execute(self, unexpanded_argv):
             self.session['command'] = expanded_arg.command
             try:
                 _validate_arguments(expanded_arg)
+            except CLIError:
```

Comment:
It passes the error along without adding it to the parser as a validation error (which would result in a usage string being pressed). Here's the output without these lines and with:

**Original**
```
az keyvault show -n fakename

usage: az keyvault show [-h] [--output {json,tsv,list,table,jsonc}]                                                                          [--verbose] [--debug] [--query JMESPATH]                                                                             [--resource-group RESOURCE_GROUP_NAME] --name                                                                        VAULT_NAME                                                                                   az keyvault show: error: The Resource 'Microsoft.KeyVault/vaults/fakename' not found within subscription 
```

**As Modified**

```
az keyvault show -n fakename

The Resource 'Microsoft.KeyVault/vaults/fakename' not found within subscription
```